config-model/src/main/java/com/yahoo/vespa/model/application/validation/BundleValidator.java

```diff
@@ -152,16 +152,18 @@ private static void validateImportedPackages(DeployLogger deployLogger, String f
                 });
     }
 
-    private void validatePomXml(DeployLogger deployLogger, String jarFilename, String pomXmlContent) {
-        try {
-            Document pom = DocumentBuilderFactory.newDefaultInstance().newDocumentBuilder()
-                    .parse(new InputSource(new StringReader(pomXmlContent)));
-            validateDependencies(deployLogger, jarFilename, pom);
-            validateRepositories(deployLogger, jarFilename, pom);
-        } catch (ParserConfigurationException e) {
-            throw new RuntimeException(e);
-        } catch (Exception e) {
-            deployLogger.log(Level.INFO, String.format("Unable to parse pom.xml from %s", jarFilename));
+    private void validatePomXml(DeployLogger deployLogger, boolean isHosted, String jarFilename, String pomXmlContent) {
+        if (isHosted) {
+            try {
+                Document pom = DocumentBuilderFactory.newDefaultInstance().newDocumentBuilder()
+                        .parse(new InputSource(new StringReader(pomXmlContent)));
+                validateDependencies(deployLogger, jarFilename, pom);
```

Comment:
Validating dependencies can make sense for non-hosted as well, so perhaps we should do that regardless?

Reply:
I'm looking into whether we can split the bundle validator into a public and internal part.